Makefile

```diff
@@ -172,14 +172,18 @@ soak-eunit: couch
 	while [ $$? -eq 0 ] ; do $(REBAR) -r eunit $(EUNIT_OPTS) ; done
 
 .PHONY: elixir
-elixir: elixir-check-formatted
+elixir: elixir-check-formatted elixir-credo
 	@rm -rf dev/lib
 	@dev/run -a adm:pass --no-eval test/elixir/run
 
 .PHONY: elixir-check-formatted
 elixir-check-formatted:
 	@cd test/elixir/ && mix format --check-formatted
 
+.PHONY: elixir-credo
+elixir-credo:
+	@cd test/elixir/ && mix deps.get && mix credo
+
```

Comment:
A concern I have is that credo is formatting the code. It shouldn't do a format - it should be doing a check, and failing if the code doesn't pass the check. This is in keeping with the philosophy that a `make check` run is a pass/fail thing, it shouldn't have side effects.

A separate target that actually does the change would be useful.

See apache/couchdb#1776 for what I did on a similar Python PR (inspired by you here!)

Reply:
@wohali Credo doesn't format code at all, and I know we are using `mix format`'s check for that 🙂